pkg/webui/console/components/pubsub-form/index.js

```diff
@@ -325,91 +333,96 @@ export default class PubsubForm extends Component {
-          title={sharedMessages.pubsubBaseTopic}
-          placeholder="base-topic"
-          component={Input}
-          required
-        />
         <Form.Field title={sharedMessages.provider} name="_provider" component={Radio.Group}>
           <Radio label="NATS" value="nats" onChange={this.handleNatsSelect} />
           <Radio label="MQTT" value="mqtt" onChange={this.handleMqttSelect} />
         </Form.Field>
-        {isMqtt ? this.mqttSection : this.natsSection}
-        <Message component="h4" content={sharedMessages.messageTypes} />
-        <Notification content={m.messageInfo} info small />
-        <Form.Field
-          name="uplink_message"
-          type="toggled-input"
-          title={sharedMessages.uplinkMessage}
-          placeholder={pathPlaceholder}
-          component={Input.Toggled}
-        />
-        <Form.Field
-          name="join_accept"
-          type="toggled-input"
-          title={sharedMessages.joinAccept}
-          placeholder={pathPlaceholder}
-          component={Input.Toggled}
-        />
-        <Form.Field
-          name="downlink_ack"
-          type="toggled-input"
-          title={sharedMessages.downlinkAck}
-          placeholder={pathPlaceholder}
-          component={Input.Toggled}
-        />
-        <Form.Field
-          name="downlink_nack"
-          type="toggled-input"
-          title={sharedMessages.downlinkNack}
-          placeholder={pathPlaceholder}
-          component={Input.Toggled}
-        />
-        <Form.Field
-          name="downlink_sent"
-          type="toggled-input"
-          title={sharedMessages.downlinkSent}
-          placeholder={pathPlaceholder}
-          component={Input.Toggled}
-        />
-        <Form.Field
-          name="downlink_failed"
-          type="toggled-input"
-          title={sharedMessages.downlinkFailed}
-          placeholder={pathPlaceholder}
-          component={Input.Toggled}
-        />
-        <Form.Field
-          name="downlink_queued"
-          type="toggled-input"
-          title={sharedMessages.downlinkQueued}
-          placeholder={pathPlaceholder}
-          component={Input.Toggled}
-        />
-        <Form.Field
-          name="location_solved"
-          type="toggled-input"
-          title={sharedMessages.locationSolved}
-          placeholder={pathPlaceholder}
-          component={Input.Toggled}
-        />
-        <Form.Field
-          name="downlink_push"
-          type="toggled-input"
-          title={sharedMessages.downlinkPush}
-          placeholder={pathPlaceholder}
-          component={Input.Toggled}
-        />
-        <Form.Field
-          name="downlink_replace"
-          type="toggled-input"
-          title={sharedMessages.downlinkReplace}
-          placeholder={pathPlaceholder}
-          component={Input.Toggled}
-        />
+        {provider === 'nats' && this.natsSection}
+        {provider === 'mqtt' && this.mqttSection}
+        {!this.useProviderDefaults && (
```

Comment:
It's going to be a nasty diff because of indentation if we don't keep it here, also preparing for when it becomes useful in TTS.

I'm also fine extracting the message types to a `messageTypesSection` and rendering that, conditional or not, that would be a one-liner.

Reply:
Extracted this to `messageTypesSection`

Reply:
Yes, good solution.